src/network/iptables.cpp

```diff
@@ -0,0 +1,284 @@
+
+    } catch (const std::exception& e) {
+        return {{}, Error(ErrorEnum::eFailed, e.what())};
+    }
+
+    return rules;
+}
+
+Error IPTables::Insert(const std::string& chain, unsigned int position, const RuleBuilder& builder)
+{
+    std::lock_guard<std::mutex> lock {mMutex};
+
+    try {
+        std::string        ruleSpec = builder.Build();
+        std::ostringstream command;
+
+        command << "iptables -t " << mTable << " -I " << chain << " " << position << " " << ruleSpec;
+
+        ExecuteCommand(command.str());
+    } catch (const std::exception& e) {
+        return Error(ErrorEnum::eFailed, e.what());
+    }
+
+    return ErrorEnum::eNone;
+}
+
+Error IPTables::DeleteRule(const std::string& chain, const RuleBuilder& builder)
+{
+    std::lock_guard<std::mutex> lock {mMutex};
+
+    try {
+        std::string        ruleSpec = builder.Build();
+        std::ostringstream command;
+
+        command << "iptables -t " << mTable << " -D " << chain << " " << ruleSpec;
+
+        ExecuteCommand(command.str());
+    } catch (const std::exception& e) {
+        return Error(ErrorEnum::eFailed, e.what());
+    }
+
+    return ErrorEnum::eNone;
+}
+
+Error IPTables::NewChain(const std::string& chain)
+{
+    std::lock_guard<std::mutex> lock {mMutex};
+
+    try {
+        std::ostringstream command;
+
+        command << "iptables -t " << mTable << " -N " << chain;
+
+        ExecuteCommand(command.str());
+    } catch (const std::exception& e) {
+        return Error(ErrorEnum::eFailed, e.what());
+    }
+
+    return ErrorEnum::eNone;
+}
+
+Error IPTables::ClearChain(const std::string& chain)
+{
+    std::lock_guard<std::mutex> lock {mMutex};
+
+    try {
+        std::ostringstream command;
+
+        command << "iptables -t " << mTable << " -F " << chain;
+
+        ExecuteCommand(command.str());
+    } catch (const std::exception& e) {
+        return Error(ErrorEnum::eFailed, e.what());
+    }
+
+    return ErrorEnum::eNone;
+}
+
+Error IPTables::DeleteChain(const std::string& chain)
+{
+    std::lock_guard<std::mutex> lock {mMutex};
+
+    try {
+        std::ostringstream command;
+
+        command << "iptables -t " << mTable << " -X " << chain;
+
+        ExecuteCommand(command.str());
+    } catch (const std::exception& e) {
+        return Error(ErrorEnum::eFailed, e.what());
+    }
+
+    return ErrorEnum::eNone;
+}
+
+RetWithError<std::vector<std::string>> IPTables::ListChains()
+{
+    std::lock_guard<std::mutex> lock {mMutex};
+
+    std::vector<std::string> chains;
+
+    try {
+        std::ostringstream command;
+
+        command << "iptables -t " << mTable << " -L -n";
+
+        auto output = ExecuteCommandWithOutput(command.str());
+
+        for (const auto& line : output) {
+            if (line.find("Chain") == 0) {
+                // Extract only the chain name (the word after "Chain")
+                size_t start = line.find("Chain") + 6;
+                size_t end   = line.find(' ', start);
```

Comment:
is it possible to have start or end equal to std::string::npos?

Reply:
no after Chain will be always chain name